utilities/src/main/java/io/syndesis/qe/validation/SfDbValidationSteps.java

```diff
@@ -134,4 +176,12 @@ private void deleteSalesforceLead(ForceApi salesforce, String firstName, String
 		final Optional<Lead> lead = queryResult.getTotalSize() > 0 ? Optional.of(queryResult.getRecords().get(0)) : Optional.empty();
 		return lead;
 	}
+
+	private Optional<Lead> getSalesforceLeadByEmail(ForceApi salesforce, String emailAddress) {
+		final QueryResult<Lead> queryResult = salesforce.query("SELECT Id,FirstName,LastName,Email,Company FROM lead where Email = '"
+				+ emailAddress + "'", Lead.class
+		);
+		final Optional<Lead> lead = queryResult.getTotalSize() > 0 ? Optional.of(queryResult.getRecords().get(0)) : Optional.empty();
```

Comment:
What about `.orElse()`? 5th paragraph https://www.mkyong.com/java8/java-8-optional-in-depth/

Reply:
In this case I don't think it would be useful, at least as far as I investigated this proposed solution. The construction above prevents the IndexOutOfBounds that could be thrown by ...getRecords().get(0). I can't find a way to utilize the .orElse().